### ISSUE TITLE
Using UIKit's UIViewController subclasses result in weird behaviour

### ISSUE THREAD

**Issue by eimantas** [schibsted/layout#17]
Version: 0.49
Xcode: 9b5
iOS Sim: 11.0

This relates to issue #15. So this is still the same XML that I have for table view controller:

```xml
<UINavigationController>
    <MyTableViewController
        title="Table">
        <UITableView
            width="100%"
            height="100%"
            rowHeight="44"
            estimatedRowHeight="44"
            >
            <UITableViewCell
                reuseIdentifier="Cell"
                rowHeight="44">
                <UILabel top="8" left="8" text="{row}" />
            </UITableViewCell>
        </UITableView>
    </MyTableViewController>
</UINavigationController>
```

The thing that bugged me is that those tiny lines in the screenshot of #15 is two tables overlapping since `<MyTableViewController>` is a subclass of `UITableViewController` and not custom subclass of  `UIViewController` conforming to `UITableViewDataSource` and `UITableViewDelegate` protocols (like in Layout's sample app).

So if I change the xml to be like this:

```xml
<UINavigationController>
    <MyTableViewController
        title="Table Foo">
        <UITableViewCell
            reuseIdentifier="Cell">
            <UILabel width="100%" height="100%" top="8" left="8" text="{row}" />
        </UITableViewCell>
    </MyTableViewController>
</UINavigationController>
```

I don't get those tiny lines and there seems to be only single table, but the instance of the controller is still `UITableViewController` and not `MyTableViewControlller`.

Here's the source of `MyTableViewController.swift`

```xml
import UIKit

import Layout

class MyTableViewController: UITableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 20
    }

    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 44.0
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let node = tableView.dequeueReusableCellNode(withIdentifier: "Cell", for: indexPath)

        node.state = [
            "row": "\(indexPath.row)"
        ]

        return node.view as! UITableViewCell
    }
}
```

So in one sentence: layout behaves weird when xml references non-direct descendants of `UIViewController` (i.e. custom subclasses of table, collection and other view controllers). The layout's sample app directly subclasses the `UIViewController` for sample tabs and conforms to needed delegates.

**Comment by nicklockwood** [comment]
Yeah, it looks like UITableViewController support is pretty broken. I need to think about the best way to approach this, but for now I'd suggest that you use a regular UIViewController subclass instead of subclassing UITableViewController.

The problem is due to the fact that UITableViewController creates its own tableView. The problem with UICollectionViewController is essentially the same.

**Comment by nicklockwood** [comment]
This is fixed in master. I'll roll it into the next release.

**Comment by nicklockwood** [comment]
Fixed in 0.4.12
